src/main/java/org/openjdk/shenandoah/RegionPopUp.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022, Amazon.com, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.shenandoah;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class RegionPopUp extends JFrame {
+    private int regionx;
+    private int regiony;
+    private int regionWidth;
+    private int regionHeight;
+    private int area;
+    private int sqSize;
+    private int cols;
+    private int regionNumber;
+    private float usedLvl;
+    private float liveLvl;
+    private float tlabLvl;
+    private float gclabLvl;
+    private float plabLvl;
+    private float sharedLvl;
+    private RegionState state;
+    private long age;
+    private RegionAffiliation affiliation;
+
+    Snapshot snapshot;
+
+    public RegionPopUp(int regionx, int regiony, Snapshot snapshot, int regionWidth, int regionHeight) {
+        this.regionx = regionx;
+        this.regiony = regiony;
+        this.snapshot = snapshot;
+        this.regionWidth = regionWidth;
+        this.regionHeight = regionHeight;
+        JPanel detailedState = new JPanel() {
+            public void paint(Graphics g) {
+                renderDetailedRegion(g);
+            }
+        };
+        this.add(detailedState);
+        area = regionWidth * regionHeight;
+        sqSize = Math.max(1, (int) Math.sqrt(1D * area / snapshot.regionCount()));
+        cols = regionWidth / sqSize;
+        regionNumber = (((regionx / sqSize) + 1) + ((regiony / sqSize) * cols)) - 1;
+//        System.out.println(regionNumber);
+        if (regionNumber >= 0 && regionNumber < snapshot.statsSize()) {
```

Comment:
Should this come out of the constructor and be in the render method? That way the popup can fetch the `RegionStat` from the latest `Snapshot` when it renders.